data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java

```diff
@@ -97,32 +81,69 @@ public ClientImpl(ConnectionStringBuilder csb, CloseableHttpClient httpClient, b
         clusterUrl = csb.getClusterUrl();
         aadAuthenticationHelper = clusterUrl.toLowerCase().startsWith(CloudInfo.LOCALHOST) ? null : TokenProviderFactory.createTokenProvider(csb, httpClient);
         clientDetails = new ClientDetails(csb.getApplicationNameForTracing(), csb.getUserNameForTracing(), csb.getClientVersionForTracing());
-        this.httpClient = httpClient;
-        this.leaveHttpClientOpen = leaveHttpClientOpen;
     }
 
     @Override
-    public KustoOperationResult execute(String command) throws DataServiceException, DataClientException {
-        return execute(DEFAULT_DATABASE_NAME, command);
+    public Mono<KustoOperationResult> executeQueryAsync(String database, String command, ClientRequestProperties properties) {
+        KustoRequest kr = new KustoRequest(command, database, properties);
+        return executeQueryAsync(kr);
     }
 
-    @Override
-    public KustoOperationResult execute(String database, String command) throws DataServiceException, DataClientException {
-        return execute(database, command, null);
+    Mono<KustoOperationResult> executeQueryAsync(@NotNull KustoRequest kr) {
+        if (kr.getCommandType() != CommandType.QUERY) {
+            kr.setCommandType(CommandType.QUERY);
+        }
+        return executeAsync(kr);
     }
 
     @Override
-    public KustoOperationResult execute(String database, String command, ClientRequestProperties properties) throws DataServiceException, DataClientException {
-        return execute(database, command, properties, determineCommandType(command));
+    public Mono<KustoOperationResult> executeMgmtAsync(String database, String command, ClientRequestProperties properties) {
+        KustoRequest kr = new KustoRequest(command, database, properties);
+        return executeMgmtAsync(kr);
     }
 
-    private KustoOperationResult execute(String database, String command, ClientRequestProperties properties, CommandType commandType)
-            throws DataServiceException, DataClientException {
-        return MonitoredActivity.invoke(
-                (SupplierTwoExceptions<KustoOperationResult, DataServiceException, DataClientException>) () -> executeImpl(database, command, properties,
-                        commandType),
-                commandType.getActivityTypeSuffix().concat(".execute"),
-                updateAndGetExecuteTracingAttributes(database, properties));
+    public Mono<KustoOperationResult> executeMgmtAsync(@NotNull KustoRequest kr) {
+        if (kr.getCommandType() != CommandType.ADMIN_COMMAND) {
+            kr.setCommandType(CommandType.ADMIN_COMMAND);
+        }
+        return executeAsync(kr);
+    }
+
+    private Mono<KustoOperationResult> executeAsync(KustoRequest kr) {
+
+        Mono<String> resultMono = executeToJsonAsync(kr)
+                .onErrorContinue((err, src) -> LOGGER.error("Error coming from src {}", src, err));
```

Comment:
Does on error continue stop the execution of further things down the pipeline? what about the rest of the function?

Reply:
onErrorContinue continues the pipeline in the event of an error. It was there for testing but I kind of want to rip all the async code out because I couldn't get it working.